inception-workload-dynamic/src/main/java/de/tudarmstadt/ukp/inception/workload/dynamic/support/AnnotationQueueOverviewDataProvider.java

```diff
@@ -300,60 +350,111 @@ public long getInProgressAmountForDocument(
 
     //Returns a random document out of all documents in the project.
     //Only a document is chosen which is not yet given to annotators more than the default number
-    //per document number
-    public SourceDocument getRandomDocument()
+    //per document number (TODO),
+    //However, if there was already a document in progress,
+    // this will always be returned (can only be one)
+
+    public SourceDocument getRandomDocument(
+        AnnotationPageBase aAnnotationPageBase, AnnotationDocument aCurrentAnno)
     {
-        //Create an empty document
-        SourceDocument document = null;
+
+        //First check for a documents that is in Progress for the user, and return this
+        for (int i = 0; i < allAnnotationDocuments.size(); i++)
```

Comment:
The random document function is used when getting a new document for a user on the annotation page, right? There we don't need to use the AnnotationQueueOverviewDataProvider. In particular, we shouldn't need to load ALL annotation documents if we are only interested in the ones for a given user and of a given state - that's what SQL queries (in DocumentService) are good for...